lib/std/fs.zig

```diff
@@ -1028,7 +1028,7 @@ pub const Dir = struct {
             .write_only => @as(u32, os.O.WRONLY),
             .read_write => @as(u32, os.O.RDWR),
         };
-        const fd = if (flags.intended_io_mode != .blocking)
+        const fd = if (builtin.zig_backend == .stage1 and flags.intended_io_mode != .blocking) // TODO
```

Comment:
`flags.intended_io_mode != .blocking` is supposed to evaluate to comptime `false` here, preventing referencing `std.event.Loop`. Are you sure we need the direct workaround? Let's look into why stage2 didn't figure out that this expression is comptime-known.

Reply:
`intended_io_mode` is coming from a non-comptime parameter here, if the condition is supposed to be comptime known then that's a bug in the function.

Reply:
Look at the type definition though:

```
pub const ModeOverride = if (is_async) Mode else enum { blocking };
```

In our case, the enum has exactly 1 tag so it should be returning true for one-possible-value.

Reply:
I see, looks like the only `io_mode` related change that was needed was to swap the order in `windows.WriteFile`. And I guess those in `os.zig` could also be made to check `std.io.mode`.

Reply:
This is a bit ugly but it does solve the problem:

```diff
--- a/lib/std/event/loop.zig
+++ b/lib/std/event/loop.zig
@@ -103,8 +103,13 @@ pub const Loop = struct {
         };
     };
 
-    var global_instance_state: Loop = undefined;
-    const default_instance: ?*Loop = switch (std.io.mode) {
+    const LoopOrVoid = switch (std.io.mode) {
+        .blocking => void,
+        .evented => Loop,
+    };
+
+    var global_instance_state: LoopOrVoid = undefined;
+    const default_instance: ?*LoopOrVoid = switch (std.io.mode) {
         .blocking => null,
         .evented => &global_instance_state,
     };

```